pytest_html/plugin.py

```diff
@@ -336,6 +336,38 @@ def _append_video(self, extra, extra_index, test_index):
             )
             self.additional_html.append(html.div(html_div, class_="video"))
 
+    class EnvironmentTable:
+        def __init__(self, config):
```

Comment:
I would split this big init up into smaller functions.

Think builder pattern.

Reply:
Not really sure to understand what to do here, I agree that everything in the init is not good but the function I can think of (generate_header, generate_row, ...) are very small... Do you want something like that?

Reply:
Yeah, I'm having trouble articulating the vision I have in my head. Give me the weekend to see if I can throw something together that you can refine. :)